actionpack/lib/abstract_controller/base.rb

```diff
@@ -16,28 +16,13 @@ def initialize(message = nil, controller = nil, action = nil)
       super(message)
     end
 
-    class Correction
-      def initialize(error)
-        @error = error
-      end
+    if defined?(DidYouMean::SpellChecker) && defined?(DidYouMean::Correctable)
+      include DidYouMean::Correctable
 
       def corrections
-        if @error.action
-          maybe_these = @error.controller.class.action_methods
-
-          maybe_these.sort_by { |n|
-            DidYouMean::Jaro.distance(@error.action.to_s, n)
-          }.reverse.first(4)
-        else
-          []
-        end
+        DidYouMean::SpellChecker.new(dictionary: controller.class.action_methods).correct(action)
       end
```

Comment:
When is the case that `DidYouMean::SpellChecker` would be defined but `DidYouMean::Correctable` would not be?  I wonder if it might be productive to wrap this logic in an extension to `StandardError`...?  Something like:

```ruby
did_you_mean do
  [action, controller.class.action_methods]
end
```